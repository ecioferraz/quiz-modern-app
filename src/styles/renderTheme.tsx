import { render, RenderResult } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export default function renderTheme(children: ReactNode): RenderResult {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}
