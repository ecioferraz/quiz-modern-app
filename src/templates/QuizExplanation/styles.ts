import styled, { css } from 'styled-components';

export const Container = styled.section`
  ${({ theme }) => css`
    h1 {
      font-weight: ${theme.font.weight.semiBold};
    }

    section:has(> img) {
      border-radius: 50%;
      height: 40px;
      width: 40px;
    }
  `}
`;
