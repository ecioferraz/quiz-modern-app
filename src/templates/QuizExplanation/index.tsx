import DetailList from '../../components/DetailList';
import TextCard from '../../components/TextCard';
import CategoryCard from '../CategoryCard';
import CategoryCardProps from '../CategoryCard/CategoryCard.props';
import * as Styled from './styles';

export default function QuizExplanation() {
  const sportDetails: CategoryCardProps[] = [
    {
      background: '#FF9F241A',
      description: '10 point for a correct answer',
      iconSrc: 'assets/icons/notebook.svg',
      title: '10 Question',
    },
    {
      background: '#FF9F241A',
      description: 'Total duration of the quiz',
      iconSrc: 'assets/icons/orange-clock.svg',
      title: '1 hour 20 min',
    },
  ];

  return (
    <Styled.Container>
      <TextCard>Brief explanation about this quiz</TextCard>
      {sportDetails.map(({ background, description, iconSrc, title }) => (
        <CategoryCard
          background={background}
          description={description}
          iconSrc={iconSrc}
          key={title}
          title={title}
        />
      ))}
      <TextCard>
        Please read the text below carefully so you can understand it
      </TextCard>
      <DetailList />
    </Styled.Container>
  );
}
