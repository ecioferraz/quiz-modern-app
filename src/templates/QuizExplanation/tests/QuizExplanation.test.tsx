import { describe, expect, it } from 'vitest';
import renderTheme from '../../../styles/renderTheme';
import QuizExplanation from '..';

describe('<QuizExplanation />', () => {
  it('should render <QuizExplanation />', () => {
    const { container } = renderTheme(<QuizExplanation />);

    expect(container).toMatchSnapshot();
  });
});
