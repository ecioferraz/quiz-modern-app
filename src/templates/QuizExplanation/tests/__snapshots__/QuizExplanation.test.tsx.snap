// Vitest Snapshot v1

exports[`<QuizExplanation /> > should render <QuizExplanation /> 1`] = `
.c6 {
  font-size: 1.4rem;
  padding-left: 1.6rem;
}

.c6 li {
  color: #666666;
  font-family: "Nunito",sans-serif;
  margin: 2.4rem 0;
}

.c6 li::marker {
  color: #333333;
}

.c6 li:last-child {
  margin-bottom: 0;
}

.c1 {
  font-size: 1.6rem;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FF9F241A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.03);
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 1.6rem 0;
}

.c2 > * {
  margin: 1.6rem;
  margin-right: 1.9rem;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
}

.c3 > section:has(img) {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 10px;
  box-shadow: 10px 24px 54px rgba(51,51,51,0.05);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.8rem;
}

.c5 h2,
.c5 p {
  margin: 0;
}

.c5 h2 {
  font-size: 1.4rem;
  font-weight: 600;
}

.c5 p {
  color: #999999;
  font-family: "Nunito",sans-serif;
  font-size: 1.2rem;
  font-weight: 400;
}

.c0 h1 {
  font-weight: 600;
}

.c0 section:has(> img) {
  border-radius: 50%;
  height: 40px;
  width: 40px;
}

<div>
  <section
    class="c0"
  >
    <h1
      class="c1"
    >
      Brief explanation about this quiz
    </h1>
    <section
      class="c2"
    >
      <section
        class="c3"
      >
        <section
          class="c4"
        >
          <img
            alt="10 Question icon"
            src="assets/icons/notebook.svg"
          />
        </section>
        <section
          class="c5"
        >
          <h2
            class="c1"
          >
            10 Question
          </h2>
          <p
            class="c1"
          >
            10 point for a correct answer
          </p>
        </section>
      </section>
    </section>
    <section
      class="c2"
    >
      <section
        class="c3"
      >
        <section
          class="c4"
        >
          <img
            alt="1 hour 20 min icon"
            src="assets/icons/orange-clock.svg"
          />
        </section>
        <section
          class="c5"
        >
          <h2
            class="c1"
          >
            1 hour 20 min
          </h2>
          <p
            class="c1"
          >
            Total duration of the quiz
          </p>
        </section>
      </section>
    </section>
    <h1
      class="c1"
    >
      Please read the text below carefully so you can understand it
    </h1>
    <ul
      class="c6"
    >
      <li>
        10 point awarded for a correct answer and no marks for a incorrect answer
      </li>
      <li>
        Tap on options to select the correct answer
      </li>
      <li>
        Click submit if you are sure you want to complete all the quizzes
      </li>
    </ul>
  </section>
</div>
`;
